src/EditorFeatures/CSharpTest/AutomaticCompletion/AutomaticLineEnderTests.cs

```diff
@@ -841,6 +846,668 @@ void Method()
+        [InlineData("class")]
+        [InlineData("struct")]
+        [InlineData("record")]
+        [InlineData("enum")]
+        [InlineData("interface")]
+        public void TestValidTypeDeclarationAndNamespace(string typeKeyword)
+        {
+            Test($@"public {typeKeyword} Bar {{}}
+$$",
+                $@"public {typeKeyword} Ba$$r {{}}");
+        }
+
+        [WpfFact]
+        public void TestMethod()
+        {
+            Test(@"
+public class Bar
+{
+    void Main()
+    {
+        $$
+    }
+}", @"
+public class Bar
+{
+    void Ma$$in()
+}");
+        }
+
+        [WpfFact]
+        public void TestValidMethodInInterface()
+        {
+            Test(@"
+public interface Bar
+{
+    void Main();
+    $$
+}", @"
+public interface Bar
+{
+    void Mai$$n();
+}");
+        }
+
+        [WpfFact]
+        public void TestValidLocalFunction()
+        {
+            Test(@"
+public class Bar
+{
+    void Main()
+    {
+        void Local()
+            $$
+        {
+        }
+    }
+}", @"
+public class Bar
+{
+    void Main()
+    {
+        void Loc$$al()
+        {
+        }
+    }
+}");
+        }
+
+        [WpfFact]
+        public void TestLocalFunction()
+        {
+            Test(@"
+public class Bar
+{
+    void Main()
+    {
+        void Local()
+        {
+            $$
+        }
+    }
+}", @"
+public class Bar
+{
+    void Main()
+    {
+        void Loca$$l()
+    }
+}");
+        }
+
+        [WpfFact]
+        public void TestIndexer1()
+        {
+            Test(@"
+public class Bar
+{
+    public int this[int i]
+    {
+        $$
+    }
+}", @"
+public class Bar
+{
+    public int thi$$s[int i]
+}");
+        }
+
+        [WpfFact]
+        public void TestIndexer2()
+        {
+            Test(@"
+public class Bar
+{
+    public int this[int i]
+    {
+        $$
+    }
+    void Main() {}
+}", @"
+public class Bar
+{
+    public int thi$$s[int i]
+    void Main() {}
+}");
+        }
+
+        [WpfFact]
+        public void TestValidIndexer()
+        {
+            Test(@"
+public class Bar
+{
+    public int this[int i]
+        $$
+    {
+    }
+}", @"
+public class Bar
+{
+    public int thi$$s[int i]
+    {
+    }
+}");
+        }
+
+        [WpfFact]
+        public void TestAddBracesForObjectCreationExpression1()
+        {
+            Test(@"
+public class Bar
+{
+    public void M()
+    {
+        var f = new Foo()
+        {
+            $$
+        };
+    }
+}
+public class Foo
+{
+    public int HH { get; set; }
+    public int PP { get; set; }
+}", @"
+public class Bar
+{
+    public void M()
+    {
+        var f = new Foo()$$
+    }
+}
+public class Foo
+{
+    public int HH { get; set; }
+    public int PP { get; set; }
+}");
+        }
+
+        [WpfFact]
+        public void TestAddBracesForObjectCreationExpression2()
+        {
+            Test(@"
+public class Bar
+{
+    public void M()
+    {
+        var f = new Foo
+        {
+            $$
+        };
```

Comment:
💭 I'm undecided on adding the `;` as part of this step. Fine to keep it or leave it for now.